editor/editor_plugin.cpp

```diff
@@ -62,7 +62,14 @@ void EditorPlugin::remove_custom_type(const String &p_type) {
 }
 
 void EditorPlugin::add_autoload_singleton(const String &p_name, const String &p_path) {
-	EditorNode::get_singleton()->get_project_settings()->get_autoload_settings()->autoload_add(p_name, p_path);
+	if (p_path.begins_with("res://")) {
+		EditorNode::get_singleton()->get_project_settings()->get_autoload_settings()->autoload_add(p_name, p_path);
+	} else {
+		const String script_path = static_cast<Ref<Script>>(get_script())->get_path();
```

Comment:
`get_script()` may be null.

While the common usage of EditorPlugin is via editor-instantiated script plugins, there is a pattern of making a temporary `EditorPlugin.new()` from a different tool script in order to access parts of the editor interface, and in these cases `get_script()` would return null.

I think an `ERR_FAIL_COND` would be sufficient to covert this case: nobody should be calling `add_autoload_singleton` from a script-less EditorPlugin instance, so it can fail in this case.

Reply:
Can a `Ref<Script>` be `nullptr`? Or should the check be fore `script.is_null()`?

Reply:
@akien-mga 
Both `script == nullptr` and `script.is_null()` will work, but `script.is_null()` is better.